app/build.gradle

```diff
@@ -339,6 +339,9 @@ android {
 
   // Set each custom apps respective source directory
   sourceSets {
+    main {
+      java.srcDirs = ['src/main/java', 'build/generated/source/kapt/kiwixRelease']
```

Comment:
This is when you try to build a custom app? Hardcoding the variant is definitely not a good idea, it is possible that you could build it but just need a file refresh? Can you switch to my branch, file>invalidate caches and restart, build and then run?

Reply:
maybe i should be more specific. the goal was to find a minimal commit to convert an android java application, here the branch "develop", so that it can accomodate a kotlin application file. to learn, i looked at branch macgills/2.5-kotlin and its commits. 

as test i converted app/KiwixApplication.java with AndroidStudio to app/KiwixApplication.kt. the reasoning was that starting from the top of the logic would need the least adjustments in other files. taking your 2.5-kotlin as base, i tried now again and pushed soloturn/2.5-kotlin.

Reply:
> @soloturn In general this is recommended to have a ticket (bug report/feature request) explaining properly the what kind of thing the PR is about.... attached to the PR. PR without ticket has proven to be more complicated to deal with.

would #1201 be ok? if not please adjust or close.

Reply:
@macgills could you do me a favour and once try if convert KiwixApplication to kotlin compiles for you, and if not what you would change in the build.gradle?